src/algebra/group_action_hom.lean

```diff
@@ -255,4 +255,33 @@ ext $ λ x, by rw [comp_apply, id_apply]
 @[simp] lemma comp_id (f : R →+*[M] S) : f.comp (mul_semiring_action_hom.id M) = f :=
 ext $ λ x, by rw [comp_apply, id_apply]
 
+variables {P : Type*} [comm_semiring P] [mul_semiring_action M P]
+variables {Q : Type*} [comm_semiring Q] [mul_semiring_action M Q]
+open polynomial
+
+/-- An equivariant map induces an equivariant map on polynomials. -/
+protected noncomputable def polynomial (g : P →+*[M] Q) : polynomial P →+*[M] polynomial Q :=
+⟨map g, λ m p, polynomial.induction_on p
+  (λ b, by rw [smul_C, map_C, coe_fn_coe, g.map_smul, map_C, coe_fn_coe, smul_C])
+  (λ p q ihp ihq, by rw [smul_add, polynomial.map_add, ihp, ihq, polynomial.map_add, smul_add])
+  (λ n b ih, by rw [smul_mul', smul_C, smul_pow, smul_X, polynomial.map_mul, map_C,
+      polynomial.map_pow, map_X, coe_fn_coe, g.map_smul, polynomial.map_mul, map_C,
+      polynomial.map_pow, map_X, smul_mul', smul_C, smul_pow, smul_X, coe_fn_coe]),
+polynomial.map_zero g, λ p q, polynomial.map_add g,
+polynomial.map_one g, λ p q, polynomial.map_mul g⟩
+
+theorem coe_polynomial (g : P →+*[M] Q) : (g.polynomial : polynomial P → polynomial Q) = map g :=
+rfl
+
 end mul_semiring_action_hom
+
+variables (M) {R'} (U : set R') [is_subring U] [is_invariant_subring M U]
+
+/-- The canonical inclusion from an invariant subring. -/
+def is_invariant_subring.subtype : U →+*[M] R' :=
```

Comment:
There has been discussion about how to call these bundled inclusions, and I'm not sure we reached any consensus. I really don't like `subtype`, because it doesn't sound like a map at all.
The discussion at https://github.com/leanprover-community/mathlib/pull/2953 seems to have settled on `subtype_hom`. Other versions in the library include `val_hom`, `coe_hom` and `incl`... we need to standardise this...

Reply:
Egō sequor `is_subring.subtype` ut videre potest

Reply:
done